mvpa2/algorithms/statis.py

```diff
@@ -8,93 +8,111 @@
-    Eigen decomposition.
-    :param C: Symmetric matrix
-    :return: eigen values, eigen vectors
+    """Return sorted and sign flipped eigen decomposition of a matrix
+
+    Parameters
+    ----------
+    C: np.ndarray
+       Symmetric matrix
+
+    Returns
+    -------
+    eigen values, eigen vectors
     """
     evals, evecs = np.linalg.eigh(h)
     # To deal with numerical errors
-    evals[evals<0] = 0
+    evals[evals < 0] = 0
     eval_desc_order = np.argsort(evals)[-1::-1]
     evecs = evecs[:, eval_desc_order]
     evals = evals[eval_desc_order]
     # Fix signs
-    evecs = evecs*np.sign(evecs[0, 0])
+    evecs = evecs * np.sign(evecs[0, 0])
     return evals, evecs
 
 
 class Statis(ClassWithCollections):
-    """
-    Implementation of alignment of subjects or tables using STATIS as presented in:
-    Abdi, H., Williams, L.J., Valentin, D., & Bennani-Dosse, M. (2012). STATIS
-    and DISTATIS: Optimum multi-table principal component analysis and three way
-    metric multidimensional scaling. Wiley Interdisciplinary Reviews:
-    Computational Statistics, 4, 124-167.
+    """Alignment of subjects or tables using STATIS
 
     This algorithm takes a list of datasets with matching samples
-    or a single dataset with `fa.tables_attr` separating subjects/tables.
-    Returns a list of `StaticProjectionMapper`s that project these feature spaces
+    or a single dataset with `fa.chunks_attr` separating subjects/tables.
+    It produces a list of `StaticProjectionMapper`s that project these feature spaces
     of input datasets into STATIS compromise space.
 
-    Note that when provided a single dataset as input, returned mappers as in the order
-    of np.unique(fa.tables_attr).
+    Note that when provided a single dataset as input, returned mappers are in
+    the order fa.chunks_attr).
 
     Statis also stores subject/table factor scores as `G_t`,
     each subject's contribution to compromise as `alpha`,
     and sample factor scores as `G_s`.
+
+    References
+    ----------
+    .. [1] Abdi, H., Williams, L.J., Valentin, D., & Bennani-Dosse, M. (2012).
+       STATIS and DISTATIS: Optimum multi-table principal component analysis and
+       three way metric multidimensional scaling. Wiley Interdisciplinary
+       Reviews: Computational Statistics, 4, 124-167.
     """
-    tables_attr = 'chunks'
-    bootstrap_iter = 0
 
-    def __init__(self, tables_attr='chunks', **kwargs):
+    chunks_attr = Parameter(
+        'chunks',
+        doc="""Name of the attribute indicating the individual chunks from
+            which a single subject/table is drawn.""")
+
+    nperms = Parameter(
+        1000, constraints=EnsureInt() & EnsureRange(min=1),
+        doc="""Number of permutations to carry out for statistical assessment.""")
+
+    def __init__(self, **kwargs):
         ClassWithCollections.__init__(self, **kwargs)
-        self.tables_attr = tables_attr
         # Subject factor scores
         self.G_t = None
```

Comment:
I believe all of those beasts should go into "ConditionalAttributes", may be even enabled by default